src/properties.jl

```diff
@@ -270,7 +270,7 @@ end
 
 get_track_order(p::Properties) = API.h5p_get_link_creation_order(p) != 0 && API.h5p_get_attr_creation_order(p) != 0
 
-function set_track_order(p::Properties, val::Bool)
+function set_track_order!(p::Properties, val::Bool)
```

Comment:
Why ? These are unexported. I've renamed it to `set_track_order!` for consistency with e.g. `set_obj_track_times!`.

Reply:
`export` does not define what is valid API or not. Non-exported functions could be public API if documented. That's why a package such as [PublicAPI.jl](https://docs.juliahub.com/PublicAPI/Jfs06/0.1.0/) exists.

While this is not documented, I would lean towards using `@deprecate` when possible. I agree this should change, but I also think we should deprecate should since `set_track_order` has existed across several releases.